src/videouplinkbandwidthpolicy/VideoUplinkBandwidthPolicy.ts

```diff
@@ -33,6 +33,11 @@ export default interface VideoUplinkBandwidthPolicy {
    */
   maxBandwidthKbps(): number;
 
+  /**
+   * Gets the factor by which to scale down the video during encoding.
+   */
+  scaleResolutionDownBy?(): number;
+
```

Comment:
Why isn't this baked into `chooseCaptureAndEncodeParameters` or `updateIndex` or something else?

Reply:
So I followed the same pattern as `maxBandwidthKbps` and create a seperate function for `scaleResolutionDownBy` which is then called by `chooseCaptureAndEncodeParameters` and `updateIndex`. I think the reason these are included in the interface originally is because we do call it in `DefaultAudioVideoController.handleHasBandwidthPriority` for active speaker case.

Reply:
Updated.